generic/tests/ping.py

```diff
@@ -58,76 +64,161 @@ def _get_loss_ratio(output):
+                                                     interface=interface,
+                                                     timeout=60)
+                    _get_loss_ratio(output)
+
+                    if status != 0:
+                        test.fail("Ping failed, status: %s,"
+                                  " output: %s" % (status, output))
+
+            # ping with exceeded packages
+            error_context.context("Ping test with exceeded package",
+                                  logging.info)
+            status, output = utils_test.ping(ip, 30, packetsize=65508,
+                                             interface=interface, timeout=60)
+            result_str = "packet size 65508 is too large. Maximum is 65507"
+            if output.find(result_str) == -1:
+                test.fail(("Ping test with exceeded package failed,"
+                           " expected string: '%s' not found" %
+                           result_str))
+            # ping with flood
+            error_context.context("Flood ping test from host",
+                                  logging.info)
+            utils_test.ping(ip, None, flood=True, output_func=None,
+                            interface=interface, timeout=flood_minutes * 60)
+
+            error_context.context("Ping test after flood ping,"
+                                  " Check if the network is still alive",
+                                  logging.info)
+            status, output = utils_test.ping(ip, counts, interface=interface,
+                                             timeout=float(counts) * 1.5)
             _get_loss_ratio(output)
 
             if status != 0:
-                raise error.TestFail("Ping failed, status: %s,"
-                                     " output: %s" % (status, output))
+                test.fail("Ping returns non-zero value %s" % output)
 
-        error.context("Flood ping test", logging.info)
-        utils_test.ping(ip, None, flood=True, output_func=None,
-                        interface=interface, timeout=flood_minutes * 60)
+    if ping_ext_host:
+        error_context.context("Ping test from guest side,"
+                              " dest: '%s'" % ext_host, logging.info)
+        pkt_sizes = packet_sizes
+        # There is no ping program for guest, so let's hardcode...
+        cmd = "ping " + ext_host  # external host
+
+        is_windows = False
+        if params.get("os_type") == "windows":
+            cmd += " -n 30 -l %s"
+            # Windows doesn't support ping with packet larger than '65500'
+            pkt_sizes = [p for p in packet_sizes if int(p) < 65500]
+            # Add a packet size just equal '65500' for windows
+            pkt_sizes.append(65500)
+            is_windows = True
+        else:
+            cmd += " -c 30 -s %s -i %s"
+
+        for size in pkt_sizes:
+            if is_windows:
+                logging.info("Ping with packet size: %s" % size)
+                #  cur_cmd = cmd % size
```

Comment:
two spaces here, and other newly commented out lines

Reply:
new patch does not have this problem, thanks for reminder